feat(discovery.azure): Add "sdk_auth" and "workload_identity" auth options (#6671)
grafana/alloy: docs/sources/reference/components/discovery/discovery.azure.md

{{< docs/alloy-config >}}

| Block                                    | Description                                        | Required |
| ---------------------------------------- | -------------------------------------------------- | -------- |
| [`managed_identity`][managed_identity]   | Managed Identity configuration for Azure API.      | no       |
| [`oauth`][oauth]                         | OAuth 2.0 configuration for Azure API.             | no       |
| [`sdk_auth`][sdk_auth]                   | Azure SDK configuration for Azure API.             | no       |
| [`tls_config`][tls_config]               | TLS configuration for requests to the Azure API.   | no       |
| [`workload_identity`][workload_identity] | Workload Identity configuration for Azure API.     | no       |

[managed_identity]: #managed_identity
[oauth]: #oauth
[sdk_auth]: #sdk_auth
[tls_config]: #tls_config
[workload_identity]: #workload_identity

{{< /docs/alloy-config >}}

> You must specify exactly one of the `oauth`, `managed_identity`, `sdk_auth`, or `workload_identity` blocks.

Author: I agree the Usage section is misleading, but I don't think we have a convention for displaying "one of these blocks is required". So I'll just leave it as it is.

| `client_secret` | `string` | OAuth 2.0 client secret. |         | yes      |
| `tenant_id`     | `string` | OAuth 2.0 tenant ID.     |         | yes      |

### `sdk_auth`

> The `sdk_auth` block configures authentication using the [Azure SDK's `DefaultAzureCredential`][default-azure-credential] chain.

Reviewer: ```suggestion
The `sdk_auth` block configures authentication with the [`DefaultAzureCredential`][default-azure-credential] chain from the Azure SDK.
```

Active voice, removes the possessive for on Azure SDK.

Should it be "configures authentication with" or "configures authentication for"? Line 108 for MEWI uses the "for" sentence structure.

Author: Thanks! I think "with" is fine!

> The `sdk_auth` block configures authentication using the [Azure SDK's `DefaultAzureCredential`][default-azure-credential] chain.

Author: No need for this
